### ISSUE TITLE
Archiving new project from scratch is not working

### ISSUE THREAD

**Issue by hsjoberg** [microsoft/react-native-macos#1917]
Hey, I'm having some problems creating an Archive inside Xcode.
I've tried creating a new project from scratch and it has the same issue.

It seems like react-native-macos is not being linked correctly.

Logging snippet from doing "Archive" in Xcode.

```
node /Users/coco/Projects/Blixt/BlixtFresh/TestMacos/node_modules/react-native-macos/cli.js bundle --entry-file index.js --platform macos --dev false --reset-cache --bundle-output /Users/coco/Library/Developer/Xcode/DerivedData/TestMacos-cqaftfzyjkkrbcbocznvjkmirhft/Build/Intermediates.noindex/ArchiveIntermediates/TestMacos-macOS/BuildProductsPath/Release/main.jsbundle --assets-dest /Users/coco/Library/Developer/Xcode/DerivedData/TestMacos-cqaftfzyjkkrbcbocznvjkmirhft/Build/Intermediates.noindex/ArchiveIntermediates/TestMacos-macOS/BuildProductsPath/Release/TestMacos.app/Contents/Resources
[...]
error Invalid platform "macos" selected.
info Available platforms are: "android", "ios". If you are trying to bundle for an out-of-tree platform, it may not be installed.
error Bundling failed.
Error: Bundling failed
    at buildBundleWithConfig (/Users/coco/Projects/Blixt/BlixtFresh/TestMacos/node_modules/@react-native-community/cli/node_modules/@react-native-community/cli-plugin-metro/build/commands/bundle/buildBundle.js:69:11)
    at buildBundle (/Users/coco/Projects/Blixt/BlixtFresh/TestMacos/node_modules/@react-native-community/cli/node_modules/@react-native-community/cli-plugin-metro/build/commands/bundle/buildBundle.js:57:10)
    at async Command.handleAction (/Users/coco/Projects/Blixt/BlixtFresh/TestMacos/node_modules/@react-native-community/cli/build/index.js:108:9)
```

Running the command `node /Users/coco/Projects/Blixt/BlixtFresh/TestMacos/node_modules/react-native-macos/cli.js bundle --entry-file index.js --platform macos --dev false --reset-cache --bundle-output /Users/coco/Library/Developer/Xcode/DerivedData/TestMacos-cqaftfzyjkkrbcbocznvjkmirhft/Build/Intermediates.noindex/ArchiveIntermediates/TestMacos-macOS/BuildProductsPath/Release/main.jsbundle --assets-dest /Users/coco/Library/Developer/Xcode/DerivedData/TestMacos-cqaftfzyjkkrbcbocznvjkmirhft/Build/Intermediates.noindex/ArchiveIntermediates/TestMacos-macOS/BuildProductsPath/Release/TestMacos.app/Contents/Resources` from the terminal gives the same error.

I also tried to hack in `"macos"` into the array in `buildBundleWithConfig()` `node_modules/@react-native-community/cli/node_modules/@react-native-community/cli-plugin-metro/build/commands/bundle/buildBundle.js:57`, but it still does not work (it then will complain about `DrawerLayoutAndroid`, which without knowing much about react-native internals, made me believe react-native-macos isn't being used and linked properly.

### Environment

```markdown
react-native -v: 11.3.6 (0.72.4 in package.json)
npm ls react-native-macos: 0.71.34 (same issue on latest main)
node -v: v20.3.1
npm -v: 9.6.7
yarn --version: 1.22.19
xcodebuild -version: Xcode 14.3 Build version 14E222b
```

### Steps to reproduce the bug

* `npx react-native@latest init TestMacos`
* `npx react-native-macos-init` (latest `main` has the same issue)
* (in order to fix unrelated Metro issues:) `yarn add metro@0.76.7`
* Open XCode and make an Archive

Note: see below for GitHub project showing the issue.

### Reproducible Demo

https://github.com/hsjoberg/RNMacosArchiveIssue

**Comment by Saadnajmi** [comment]
Did this work for you in an earlier version? Trying to understand if this is a recent regression.

**Comment by hsjoberg** [comment]
Yes, this has worked for me before. I've been Archiving and publishing app builds for ~1.5 years.

But due to the general difficulties moving up to >= 0.70, I recently started a new react-native & react-native-macos project from scratch and added back all the code for my app to that shell.

**Comment by Saadnajmi** [comment]
So to confirm, you used to be able to archive with React Native macOS pre 0.70, so probably 0.68?

**Comment by hsjoberg** [comment]
Yes, I used to be able to archive with React Native macOS.

Though instead of going with 0.68 I stayed up-to-date with latest main branch by pointing to specific commits.
https://github.com/hsjoberg/blixt-wallet/pull/1267/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L123

It seems like something in the template broke at some point, but not sure.
Are you able to reproduce by following the steps or by building the demo project?
